### PR TITLE
Created Better async Event Wrapping

### PR DESCRIPTION
**Pull request by clicketyclackety**
# Description

Many of the server notifications are asynchronous methods being run without await etc.
Not only that but currently the Rhino events are being wrapped 1 for 1.

This PR does a couple of things to improve this state of affairs
1. It wraps the Rhino Events with more coherent Crash ones. e.g. Deselect All is just a Deselect event
2. UnDelete has a flag in it as I believe UnDelete caused server sync issues
	> In short, UnDelete was sending new in-place geometry to the server without clearing the transform. This PR does not explicitly fix it (that's not its purpose) but it does add a flag into the appropriate event args so that this can be caught
3. All events that subscribe to Rhino Events are now done inside an `EventWrapper`
4. This `EventWrapper` also avoids any async void problems and converts them into nicely legible async events
5. This PR also (, as I was rummaging around in events,) adds the UpdateEventArgs for Object Attributes.
6. 

_Good Reading_
https://medium.com/@a.lyskawa/the-hitchhiker-guide-to-asynchronous-events-in-c-e9840109fb53

Fixes # (issue)

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Uhh....

- [ ] Test A
- [ ] Test B

# Checklist:

- [x] My code follows the style of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the docs
- [x] My changes generate no new warnings

### CODE REVIEW

File: src/Crash.Handlers/Utils/RhinoObjectAttributesUtils.cs [hunk 1]
```diff
@@ -0,0 +1,191 @@
+﻿using Rhino.DocObjects;
+
+namespace Crash.Handlers
+{
+	public static class RhinoObjectAttributesUtils
+	{
+		// TODO : Move
+		internal static Dictionary<string, string> GetAttributeDifferencesAsDictionary(ObjectAttributes oldAttributes,
```
Comment (Lukasfuh) [inline src/Crash.Handlers/Utils/RhinoObjectAttributesUtils.cs:8]: This is very ugly. Definetly needs to be improved. Fine to do it in a separate PR

**Review by Lukasfuh — APPROVED** [review]
Very nice! This is improving the events a looot!

File: src/Crash.Handlers/Utils/RhinoObjectAttributesUtils.cs (hunk 1 above)
Comment (clicketyclackety) [reply src/Crash.Handlers/Utils/RhinoObjectAttributesUtils.cs:8]: Agreed haha
